gcloud/bigtable/test_client.py

```diff
@@ -31,18 +175,53 @@ def _getTargetClass(self):
     def _makeOne(self, *args, **kwargs):
         return self._getTargetClass()(*args, **kwargs)
 
+    def _makeOneWithMocks(self, *args, **kwargs):
+        from gcloud._testing import _Monkey
+        from gcloud.bigtable import client as MUT
+
+        mock_make_data_stub = _MakeStubMock()
+        mock_make_instance_stub = _MakeStubMock()
+        mock_make_operations_stub = _MakeStubMock()
+        mock_make_table_stub = _MakeStubMock()
+        with _Monkey(MUT, _make_data_stub=mock_make_data_stub,
+                     _make_instance_stub=mock_make_instance_stub,
+                     _make_operations_stub=mock_make_operations_stub,
+                     _make_table_stub=mock_make_table_stub):
+            return self._makeOne(*args, **kwargs)
+
     def _constructor_test_helper(self, expected_scopes, creds,
                                  read_only=False, admin=False,
                                  user_agent=None, timeout_seconds=None,
                                  expected_creds=None):
+        from gcloud._testing import _Monkey
         from gcloud.bigtable import client as MUT
 
         user_agent = user_agent or MUT.DEFAULT_USER_AGENT
         timeout_seconds = timeout_seconds or MUT.DEFAULT_TIMEOUT_SECONDS
-        client = self._makeOne(project=self.PROJECT, credentials=creds,
-                               read_only=read_only, admin=admin,
-                               user_agent=user_agent,
-                               timeout_seconds=timeout_seconds)
+
+        mock_make_data_stub = _MakeStubMock()
+        mock_make_instance_stub = _MakeStubMock()
+        mock_make_operations_stub = _MakeStubMock()
+        mock_make_table_stub = _MakeStubMock()
+        with _Monkey(MUT, _make_data_stub=mock_make_data_stub,
+                     _make_instance_stub=mock_make_instance_stub,
+                     _make_operations_stub=mock_make_operations_stub,
+                     _make_table_stub=mock_make_table_stub):
+            client = self._makeOne(project=self.PROJECT, credentials=creds,
+                                   read_only=read_only, admin=admin,
+                                   user_agent=user_agent,
+                                   timeout_seconds=timeout_seconds)
+
+        # Verify the mocks.
+        self.assertEqual(mock_make_data_stub.calls, [client])
+        if admin:
+            self.assertEqual(mock_make_instance_stub.calls, [client])
```

Comment:
It's definitely a very fine and subtle concern. I see value in it, but I have not yet persuaded the rest of the world. :-)